src/network/networkd-state-file.c

```diff
@@ -200,6 +201,22 @@ int manager_save(Manager *m) {
                                 return r;
                 }
 
+                if (link->network->dhcp_use_captive_portal) {
+                        r = sd_dhcp_lease_get_captive_portal(link->dhcp_lease, &captive_portal);
+                        if (r < 0 && r != -ENODATA)
+                                return r;
+                }
+
+                if (!captive_portal && link->network->dhcp6_use_captive_portal) {
+                        r = sd_dhcp6_lease_get_captive_portal(link->dhcp6_lease, &captive_portal);
+                        if (r < 0 && r != -ENODATA)
+                                return r;
+                }
+
+                if (!captive_portal && link->network->ipv6_accept_ra_use_captive_portal) {
+                        captive_portal = link->ndisc_captive_portal;
+                }
```

Comment:
See https://www.rfc-editor.org/rfc/rfc8910.html#name-precedence-of-api-uris.
Please check captive portals obtained through multiple protocols are identical, and log that captive portal provided by DHCPv4 has higher precedence if they are not identical.

Reply:
Done.